halo2_proofs/src/plonk/circuit.rs

```diff
@@ -484,13 +558,23 @@ pub trait Circuit<F: Field> {
     fn synthesize(&self, config: Self::Config, layouter: impl Layouter<F>) -> Result<(), Error>;
 }
 
+/// A placeholder which will be resolved as a fixed column during optimization.
+#[cfg_attr(feature = "unstable-dynamic-lookups", derive(Clone, Copy, Debug))]
+#[cfg(feature = "unstable-dynamic-lookups")]
+pub struct VirtualColumn(pub(crate) DynamicTable);
```

Comment:
The idea of a "virtual column" seems like it should be more general. Actually I'm not sure how this type is used; is it for the tag columns?

Reply:
Yes, this should be renamed this to `TableTag`.